src/backend/distributed/planner/distributed_planner.c

```diff
@@ -906,6 +906,24 @@ CreateDistributedPlan(uint64 planId, bool allowRecursivePlanning, Query *origina
 
 		EnsurePartitionTableNotReplicated(targetRelationId);
 
+		/*
+		 * We support a limited set of INSERT .. SELECT queries if the query
+		 * references a null-dist-key table.
+		 *
+		 * Note that the planner specific restrictions that apply to INSERT ..
+		 * SELECT queries will be enforced for such queries too, later in INSERT
+		 * .. SELECT planners.
+		 */
+		if (CheckInsertSelectQuery(originalQuery))
```

Comment:
note that this code path has changed quite a bit recently so this logic needs a rebase/rewrite

Reply:
Yeah, I was planning to do the same but was waiting for https://github.com/citusdata/citus/pull/6792 to get merged in too. Anyway, let me do a rebase now.